gen3-cli/cot/backend/create/template/set_context.py

```diff
@@ -0,0 +1,241 @@
+import os
+import boto3
+import json
+import tempfile
+from cot.backend.common.fsutils import Search
+from .context_tree import (
+    find_gen3_root_dir,
+    find_gen3_dirs
+)
+from .environment import Environment
+
+
+def set_context(
+    cwd=None,
+    environment_obj=None
+):
+    e = environment_obj  # shortcut
+    l = Environment()  # local env
+    if e.GENERATION_CONTEXT_DEFINED:
+        return
+    e.GENERATION_CONTEXT_DEFINED = True
+    l.GENERATION_CONTEXT_DEFINED_LOCAL = True
+    if not e.GENERATION_TMPDIR:
+        # *******************************************
+        # * TODO: decide how to do temp dir cleanup *
+        # *******************************************
+        e.GENERATION_TMPDIR = tempfile.mkdtemp()
```

Comment:
if temp dir is located somewhere in the /tmp folder (or in the docker) we don't need to cleanup it, it's enough just to make temp directory containing some unique value (like /tmp/{randomstr}_codeontap). /tmp will be cleared on reboot, and insides of the docker container will be dropped when it's re-created.

Reply:
Ye, you're right.